src/crate.ts

```diff
@@ -49,6 +50,18 @@ export interface QueryOptions {
 
 const defaultConfig = { sqlEndpoint: 'http://localhost:4200/_sql' }
 
+const retryOptions: Options = {
+  minTimeout: ms('30s'),
+  maxTimeout: ms('1h'),
+  // This translates to retrying for up to 24 hours. It takes 7 retries
+  // to reach the maxTimeout of 1 hour.
+  retries: 30,
```

Comment:
I checked the math, and if I'm doing this right (multiply x2 for each retry), we would actually hit 24 hours between the 12th and 13th retry:

```
Retry 8: 1 hour, 4 minutes
Retry 9: 2 hours, 8 minutes
Retry 10: ~4 hours
Retry 11: ~8-1/2 hours
Retry 12: ~17 hours
Retry 13: ~34 hours
```

Reply:
`maxTimeout` is `1h`

Reply:
Aha! I was confusing myself. So at retry 7, we hit 1 hour, and then after that, we retry every hour, up to 23 more times, which would put the last retry at 24 hours. 👍

Reply:
This does somewhat go against the spirit of exponential retries (after retry 7, it isn't exponential anymore), which is something we need to be aware of. We sync 28 collections, so if something is going catastrophically, wrong, this means that once per hour, 28 separate sync processes are all still hitting the database.

I think that's sort of reasonable in this case, because 28 database operations isn't too many, but we should just keep in mind that this wouldn't scale if we had a lot more collections that we were trying to sync.

If we were operating at a higher scale (syncing many more collections), I would suggest that we add jitter (p-retry has an easy option for this, `randomize: true`) and re-work this so that it's still exponential and capped at 12 or 13 retries so that it works out to no longer retrying once it's been 24 hours.

Reply:
I agree -- just thinking out loud. I'm all for incremental improvements. 🙂

Reply:
Also, `randomize` multiples the timeout by multiplying by a number between 1 and 2, which is potentially a non-trivial additional amount of time.

Reply:
It's a little unclear from [the explanation of the options](https://github.com/tim-kos/node-retry?tab=readme-ov-file#retrytimeoutsoptions), but I've been interpreting it as "by default, multiply by 2 on each retry; if `randomize: true` is set, then multiply by a number between 1 and 2 instead"

Reply:
That's interesting.